src/t_string.c

```diff
@@ -593,15 +596,15 @@ void incrDecrCommand(client *c, long long incr) {
     } else {
         new = createStringObjectFromLongLongForValue(value);
         if (o) {
-            dbReplaceValue(c->db,c->argv[1],new);
+            dbReplaceValueWithDictEntry(c->db,c->argv[1],new,de);
         } else {
             dbAdd(c->db,c->argv[1],new);
         }
     }
     signalModifiedKey(c,c->db,c->argv[1]);
     notifyKeyspaceEvent(NOTIFY_STRING,"incrby",c->argv[1],c->db->id);
     server.dirty++;
-    addReplyLongLong(c, value);
+    addReplyLongLongFromStr(c,new);
```

Comment:
this change can cause use-after-free in the `tests/unit/moduleapi/keyspace_events.tcl` test, when the test uses RM_StringDMA to modify the value from within the KSN, and this command uses it after it was released.
the reason it doesn't happen is because the eviction policy allows for shared integers, so despite "releasing" this object from within the KSN. it doesn't free the memory.

Reply:
it can be solved by moving the reply a few lines upwards.
we can also argue that the module is violating redis, but it used to work so far (specifically for INCR), and the optimization broke it, so shall we fix that?
WDYT?